src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -2406,15 +2413,26 @@ def security_domain_download(cmd, client, hsm_name, sd_wrapping_keys, security_d
 
         certificates.append(sd_jwk)
 
+    # save security-domain backup value to local file
+    def _save_to_local_file(file_path, security_domain):
+        try:
+            with open(file_path, 'w') as f:
+                f.write(security_domain.value)
+        except:  # pylint: disable=bare-except
+            if os.path.isfile(file_path):
+                os.remove(file_path)
+
     ret = client.download(
         vault_base_url=hsm_name or vault_base_url,
         certificates=CertificateSet(certificates=certificates, required=sd_quorum)
     )
 
-    try:
-        with open(security_domain_file, 'w') as f:
-            f.write(ret.value)
-    except:  # pylint: disable=bare-except
-        if os.path.isfile(security_domain_file):
-            os.remove(security_domain_file)
+    if not no_wait:
+        polling_ret = _wait_security_domain_operation(client, hsm_name, 'download')
+        # Due to service defect, status could be 'Success' or 'Succeeded' when it succeeded
+        if polling_ret and getattr(polling_ret, 'status', None) != 'Failed':
```

Comment:
pls lower/upper case then compare on string

Reply:
I prefer we rely on service in avoid of risks in type convertion.